### ISSUE TITLE
`float` is not precise enough for UTM coordinates.

### ISSUE THREAD

**Issue by VRichardJP** [autowarefoundation/autoware.universe#3090]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

Autoware can load maps using UTM coordinate system, yet it stores a lot of xyz data using `float` type.
The problem is that `float` is not precise enough to manipulate UTM coordinates.

Here is a simple example:
Tokyo station is located at `lat=35.6785133, lon=139.763416`
With this [converter](https://www.earthpoint.us/Convert.aspx), I get:
- MGRS local coordinates: `x=88096, y=48996`
- UTM local coordinates: `x=388096, y=3948996`

According to this [IEEE-754 float converter](https://www.h-schmidt.net/FloatConverter/IEEE754.html):
- The next representable `float` value after `88096.0` is `88096.0078125`
- The next representable `float` value after `3948996.0` is `3948996.25`

In other words, if I use MGRS coordinate system around Tokyo, coordinate `float` precision is 7 millimeter. If I use UTM, `float` is only 25 centimeter precise...

One could argue that 7 millimeters is not that precise when using MGRS. Certainly, everyone will agree that 25 centimeters is unacceptable. Unless you are lucky enough to be somewhere with low UTM coordinates, it is basically suicide to use Autoware with UTM maps.

I don't see many solutions:
- either Autoware should drop UTM support
- or Autoware should use `double` type everywhere (in such case we would be nm precise even in the worst UTM cases)

### Expected behavior

Autoware is precise enough.

### Actual behavior

Not precise enough if UTM are used.

### Steps to reproduce

N/A

**Comment by mitsudome-r** [comment]
`float` issue is already known issue in TIER IV.
Since there are many source codes that assumes float at the moment (like PCL), TIER IV decided to use MGRS.
If people prefer to use other coordinate, they can select UTM projector + origin to assign their preferred origin to keep the converted numbers low enough to keep their required precision.
https://github.com/autowarefoundation/autoware.universe/blob/main/map/map_loader/config/lanelet2_map_loader.param.yaml

**Comment by mitsudome-r** [comment]
I'm closing the issue that was inactive for a long time. Feel free to reopen if the issue is still relevant
